[L4D2] Update 'IEngineTool' and 'IBaseClientDLL'

A1mDev opened the pull request:
Some stuff copied from 'hl2sdk-csgo' and tested. Used for the [sourcetvsupport-client](https://github.com/A1mDev/sourcetvsupport-client) repository. This repository contains several fixes for the game Left4Dead2 so that the client can view demos or watch streams from SourceTV. The client allows us to view demos and watch streams from SourceTV in an -insecure mode.

Kenzzer commented:
Considering the 2 PR you opened have the same scope why not combine the two changes ?

A1mDev commented:
Latest changes: 
1) Renamed header file name 'gameeventlistener' to 'GameEventListener', linux compiler is case sensitive.
2) Removed unused 'imovehelper.h' header file in file 'usercmd.h', because the very incude of this file requires a few more header files.
